res/values/strings.xml

```diff
@@ -696,33 +693,27 @@
-    <string name="RegistrationActivity_you_must_specify_your_country_code">You must specify your
-        country code
-    </string>
-    <string name="RegistrationActivity_you_must_specify_your_phone_number">You must specify your
-        phone number
-    </string>
+    <string name="RegistrationActivity_you_must_specify_your_country_code">You must specify your country code</string>
+    <string name="RegistrationActivity_you_must_specify_your_phone_number">You must specify your phone number</string>
     <string name="RegistrationActivity_invalid_number">Invalid number</string>
-    <string name="RegistrationActivity_the_number_you_specified_s_is_invalid">The number you
-        specified (%s) is invalid.
-    </string>
+    <string name="RegistrationActivity_the_number_you_specified_s_is_invalid">The number you specified (%s) is invalid.</string>
     <string name="RegistrationActivity_missing_google_play_services">Missing Google Play Services</string>
     <string name="RegistrationActivity_this_device_is_missing_google_play_services">This device is missing Google Play Services. You can still use Signal, but this configuration may result in reduced reliability or performance.\n\nIf you are not an advanced user, are not running an aftermarket Android ROM, or believe that you are seeing this in error, please contact support@signal.org for help troubleshooting.</string>
     <string name="RegistrationActivity_i_understand">I understand</string>
-    <string name="RegistrationActivity_play_services_error">Play Services Error</string>
+    <string name="RegistrationActivity_play_services_error">Google Play Services Error</string>
     <string name="RegistrationActivity_google_play_services_is_updating_or_unavailable">Google Play Services is updating or temporarily unavailable. Please try again.</string>
     <string name="RegistrationActivity_terms_and_privacy">Terms &amp; Privacy Policy</string>
     <string name="RegistrationActivity_no_browser">Unable to open this link. No web browser found.</string>
     <string name="RegistrationActivity_more_information">More information</string>
     <string name="RegistrationActivity_less_information">Less information</string>
-    <string name="RegistrationActivity_signal_needs_access_to_your_contacts_and_media_in_order_to_connect_with_friends">Signal needs access to your contacts and media in order to connect with friends, exchange messages, and make secure calls</string>
+    <string name="RegistrationActivity_signal_needs_access_to_your_contacts_and_media_in_order_to_connect_with_friends">Signal requires permission to view your contacts, in order to send messages and initiate calls to your contacts. Signal also requires permission to access media on your device in order to add files as attachments. Finally, Signal requires permission to view your call status, so Signal calls won\'t interfere with other calls.</string>
```

Comment:
From your proposal:
"Signal needs permission to read your contacts" "[to] make secure calls". That is not true. It only needs this permission to call to known contacts without having to type the number, but not to make calls in general.

"permission" is a keyword and should not be omitted.

I understand it seems verbose, but that is to stay factual and to form proper sentences.

Reply:
So the purpose of this string is to inform a user why a certain permission is needed. In my opinion that needs to be explained in detail, because a user should get enough information to really understand why.

After this text promp, Signal will request 5 permissions:
https://github.com/signalapp/Signal-Android/blob/0ef01cc620cc907da6078c93cc97d31e6612b3f3/app/src/main/java/org/thoughtcrime/securesms/registration/fragments/WelcomeFragment.java#L87

- WRITE_CONTACTS
- READ_CONTACTS
- WRITE_EXTERNAL_STORAGE
- READ_EXTERNAL_STORAGE
- READ_PHONE_STATE

But Android simplifies that and only shows the user 3 popups:
- Allow Signal to access your contacts?
- Allow Signal to access photos, media and files on your device?
- Allow Signal to make and manage phone calls?

Now this causes an annoying issue: if we speak about "access your contacts" it is simply not really clear to users what exactly is meant by "to access". What it really means is both read and write. Those are two permissions, and thus should come with two explanations.

In total we should explain why Signal needs 5 permissions.

"permission to view your address book" is very clear, but should in fact be "permission to read and write your address book" (one permission for two use cases)

You can also see that Android simplifies READ_PHONE_STATE to "make and manage phone calls" which is not at all true. Quite annoying. How can we factually inform a user if Android throws in some ambiguous popups? I really dislike what Android is doing here. It should be considered a bug in Android.

Reply:
Ok to "permission to read and write your address book".